### PR TITLE
fix Issue 22252 - ImportC: Array, Function parameter types should be …

### PR DESCRIPTION
**Pull request by WalterBright**
…converted to pointers

### CODE REVIEW

File: src/dmd/typesem.d [hunk 1]
```diff
@@ -1336,6 +1336,29 @@ extern(C++) Type typeSemantic(Type type, const ref Loc loc, Scope* sc)
...
                 Type t = fparam.type.toBasetype();
 
+                if (sc.flags & SCOPE.Cfile)
+                {
+                    /* C11 6.7.6.3-7 array of T is converted to pointer to T
+                     */
+                    if (auto ta = t.isTypeDArray())
+                    {
+                        t = ta.next.pointerTo();
+                        fparam.type = t;
```
Comment (ibuclaw) [inline src/dmd/typesem.d:1346]: What about implicit conversion for callers? Don't they need to be handled explicitly too?

Comment (WalterBright) [reply src/dmd/typesem.d:1346]: Yes, but it's completely different. This PR is about adjusting the *type* of the function parameter. You're talking about doing a *conversion* of an expression. It's covered by different paragraphs in the spec. I plan on the next PR being about that.

**Review by ibuclaw — APPROVED** [review]

**Comment by ibuclaw** [comment]
Looks like the forking gc tests fail on OSX64
```
Testing forkgc2 timelimit -t 10 ./generated/osx/release/64/forkgc2 /Users/braddr/sandbox/at-client/pull-4967297-Darwin_64_64/dmd/generated/osx/release/64/dmd -m64 -fPIC -w -I../../src -I../../import -Isrc -defaultlib= -debuglib= -dip1000 -L-lpthread -L-lm -L/Users/braddr/sandbox/at-client/pull-4967297-Darwin_64_64/druntime/generated/osx/debug/64/libdruntime.a -g -debug -ofgenerated/osx/debug/64/alloc_from_assert src/alloc_from_assert.d forkgc2(39939,0x700000187000) malloc: *** error for object 0x7ffe02403130: double free *** set a breakpoint in malloc_error_break to debug
```

**Comment by rainers** [comment]
> Looks like the forking gc tests fail on OSX64

forkgc2 is not run with the forking GC, but tests calling fork() with the regular GC. I suspect that the failures are spurious.

**Comment by ibuclaw** [comment]
@WalterBright can you rebase?

**Pull request merged**